# Homework-01
#### Name: Muhammad Talha Salani
#### ID: ms07725
#### Email: ms07725@st.habib.edu.pk

## Part-1 (40 points)

###### Format the 580SurveyCleanup2.csv dataset using Python. See the Week2-Class DataWrangling.ipynb file for starter code.

In [210]:
import pandas as pd
import numpy as np
import matplotlib as plt

In [211]:
df = pd.read_csv('580SurveyCleanup2.csv', header = 1)

In [212]:
df

No.             Time      Status  Duration (Seconds)  Access Code  \
0     1  10/3/2023 17:49   Completed               11646          NaN   
1     2  10/3/2023 17:49   Completed                 113          NaN   
2     3  10/3/2023 17:49   Completed                 572          NaN   
3     4  10/3/2023 17:50   Completed                 365          NaN   
4     5  10/3/2023 17:51   Completed                  92          NaN   
5     6  10/3/2023 17:59  Incomplete                 221          NaN   
6     7  10/3/2023 18:23   Completed              115983          NaN   
7     8  10/3/2023 19:10  Incomplete                  40          NaN   
8     9  10/3/2023 19:25   Completed                 137          NaN   
9    10  10/3/2023 19:56  Incomplete                 179          NaN   
10   11  10/3/2023 20:14   Completed                 127          NaN   
11   12  10/3/2023 21:38   Completed                 269          NaN   
12   13  10/3/2023 21:41   Completed                 117          NaN   
13   14  10/3/2023 22:49   Completed                5966          NaN   
14   15  10/3/2023 23:25   Completed                 254          NaN   
15   16   10/4/2023 0:28   Completed                 526          NaN   
16   17   10/4/2023 5:22   Completed                 398          NaN   
17   18   10/4/2023 5:36   Completed              128253          NaN   
18   19   10/4/2023 5:47   Completed                 239          NaN   
19   20  10/4/2023 10:32   Completed                 255          NaN   
20   21  10/4/2023 11:03   Completed                 256          NaN   
21   22  10/4/2023 12:34   Completed                 674          NaN   
22   23  10/4/2023 13:36   Completed                 130          NaN   
23   24  10/4/2023 13:50   Completed                 275          NaN   
24   25  10/4/2023 21:28   Completed                 325          NaN   
25   26   10/5/2023 9:37   Completed                 257          NaN   
26   27   10/5/2023 9:36   Completed                 657          NaN   
27   28  10/5/2023 10:36   Completed                1750          NaN   
28   29  10/5/2023 11:02   Completed                 887          NaN   
29   30  10/5/2023 11:17   Completed                 180          NaN   
30   31  10/5/2023 12:52   Completed                 392          NaN   
31   32  10/5/2023 12:56  Incomplete                 192          NaN   
32   33  10/5/2023 13:52   Completed                 277          NaN   
33   34  10/5/2023 15:34   Completed                 119          NaN   
34   35  10/5/2023 15:41   Completed                 149          NaN   
35   36  10/5/2023 17:58   Completed                 285          NaN   

    Email Address       IP Address        Country    Region  \
0             NaN   100.36.167.226  United States  Virginia   
1             NaN    98.169.26.134  United States  Virginia   
2             NaN     98.169.52.23  United States  Virginia   
3             NaN   69.255.182.253  United States  Virginia   
4             NaN   138.88.108.230  United States       NaN   
5             NaN    108.56.134.80  United States  Virginia   
6             NaN   108.48.175.220  United States  Virginia   
7             NaN    173.67.199.85  United States  Virginia   
8             NaN     74.96.114.12  United States  Virginia   
9             NaN    107.77.241.32  United States     Texas   
10            NaN   138.88.182.245  United States  Maryland   
11            NaN     72.83.246.60  United States  Virginia   
12            NaN    96.241.43.105  United States  Virginia   
13            NaN    71.191.72.223  United States  Virginia   
14            NaN    73.134.53.227  United States  Maryland   
15            NaN     73.86.219.52  United States  Maryland   
16            NaN   69.138.215.127  United States  Virginia   
17            NaN    72.196.231.70  United States  Virginia   
18            NaN     100.15.54.17  United States  Virginia   
19            NaN   184.170.83.239  United

In [213]:
column_mapping = {
    '002': 'Q1.002',
    '004': 'Q1.004',
    'Male': 'Q2.Male',
    'Female': 'Q2.Female',
    'NonBinary': 'Q2.NonBinary',
    'Prefer not to answer': 'Q2.PreferNotToAnswer',
    'Age (years)': 'Q3.Age',
    'Height (Inches)': 'Q4.Height',
    'Country of Citizenship': 'Q5.CountryOfCitizenship',
    'Undergraduate Degree': 'Q6.UndergradDegree',
    'Expected Graduation date from Mason MS program?': 'Q7.ExpectedGraduation',
    'Microsoft/Windows': 'Q8.MicrosoftWindows',
    'Apple/MacBook': 'Q8.AppleMacBook',
    'Other': 'Q8.Other',
    'Yes, Full Time': 'Q9.YesFullTime',
    'Working, but not Full Time': 'Q9.WorkingButNotFullTime',
    'Not Working while attending Mason': 'Q9.NotWorking',
    'Little/none': 'Q10.Little/None',   
    'Some familiarity': 'Q10.SomeFamiliarity',  
    'Average user': 'Q10.AverageUser',  
    'Frequent use for projects': 'Q10.FrequentUseForproj',
    'Fluent/expert': 'Q10.FluentExpert',
    
    'Little/none.1' : 'Q11.Little/none',
    'Some familiarity.1' : 'Q11.Some familiarity',
    'Average user.1' : 'Q11.Average user',
    'Frequent use for projects.1' : 'Q11.Frequent use for projects',
    'Fluent/expert.1' : 'Q11.Fluent/expert',

    'Little/none.2' : 'Q12.Little/none',
    'Some familiarity.2' : 'Q12.Some familiarity',
    'Average user.2' : 'Q12.Average user',
    'Frequent use for projects.2' : 'Q12.Frequent use for projects',
    'Fluent/expert.2' : 'Q12.Fluent/expert',
    
    'Yes': 'Q13.Yes',
    'No': 'Q13.No',
    'Yes.1' : 'Q14.Yes',
    'No.1' : 'Q14.No',
    
    'Not sure': 'Q14.NotSure',
    'What is your goal for learning about data analytics?': 'Q15.GoalForLearningDataAnalytics'
}

df.columns = [column_mapping.get(col, col) for col in df.columns]

In [214]:
df

No.             Time      Status  Duration (Seconds)  Access Code  \
0     1  10/3/2023 17:49   Completed               11646          NaN   
1     2  10/3/2023 17:49   Completed                 113          NaN   
2     3  10/3/2023 17:49   Completed                 572          NaN   
3     4  10/3/2023 17:50   Completed                 365          NaN   
4     5  10/3/2023 17:51   Completed                  92          NaN   
5     6  10/3/2023 17:59  Incomplete                 221          NaN   
6     7  10/3/2023 18:23   Completed              115983          NaN   
7     8  10/3/2023 19:10  Incomplete                  40          NaN   
8     9  10/3/2023 19:25   Completed                 137          NaN   
9    10  10/3/2023 19:56  Incomplete                 179          NaN   
10   11  10/3/2023 20:14   Completed                 127          NaN   
11   12  10/3/2023 21:38   Completed                 269          NaN   
12   13  10/3/2023 21:41   Completed                 117          NaN   
13   14  10/3/2023 22:49   Completed                5966          NaN   
14   15  10/3/2023 23:25   Completed                 254          NaN   
15   16   10/4/2023 0:28   Completed                 526          NaN   
16   17   10/4/2023 5:22   Completed                 398          NaN   
17   18   10/4/2023 5:36   Completed              128253          NaN   
18   19   10/4/2023 5:47   Completed                 239          NaN   
19   20  10/4/2023 10:32   Completed                 255          NaN   
20   21  10/4/2023 11:03   Completed                 256          NaN   
21   22  10/4/2023 12:34   Completed                 674          NaN   
22   23  10/4/2023 13:36   Completed                 130          NaN   
23   24  10/4/2023 13:50   Completed                 275          NaN   
24   25  10/4/2023 21:28   Completed                 325          NaN   
25   26   10/5/2023 9:37   Completed                 257          NaN   
26   27   10/5/2023 9:36   Completed                 657          NaN   
27   28  10/5/2023 10:36   Completed                1750          NaN   
28   29  10/5/2023 11:02   Completed                 887          NaN   
29   30  10/5/2023 11:17   Completed                 180          NaN   
30   31  10/5/2023 12:52   Completed                 392          NaN   
31   32  10/5/2023 12:56  Incomplete                 192          NaN   
32   33  10/5/2023 13:52   Completed                 277          NaN   
33   34  10/5/2023 15:34   Completed                 119          NaN   
34   35  10/5/2023 15:41   Completed                 149          NaN   
35   36  10/5/2023 17:58   Completed                 285          NaN   

    Email Address       IP Address        Country    Region  \
0             NaN   100.36.167.226  United States  Virginia   
1             NaN    98.169.26.134  United States  Virginia   
2             NaN     98.169.52.23  United States  Virginia   
3             NaN   69.255.182.253  United States  Virginia   
4             NaN   138.88.108.230  United States       NaN   
5             NaN    108.56.134.80  United States  Virginia   
6             NaN   108.48.175.220  United States  Virginia   
7             NaN    173.67.199.85  United States  Virginia   
8             NaN     74.96.114.12  United States  Virginia   
9             NaN    107.77.241.32  United States     Texas   
10            NaN   138.88.182.245  United States  Maryland   
11            NaN     72.83.246.60  United States  Virginia   
12            NaN    96.241.43.105  United States  Virginia   
13            NaN    71.191.72.223  United States  Virginia   
14            NaN    73.134.53.227  United States  Maryland   
15            NaN     73.86.219.52  United States  Maryland   
16            NaN   69.138.215.127  United States  Virginia   
17            NaN    72.196.231.70  United States  Virginia   
18            NaN     100.15.54.17  United States  Virginia   
19            NaN   184.170.83.239  United

In [215]:
df.to_csv("formatted_output_Part_1.csv", index=False, header=True)

## Part-2 (30 points)

In [217]:
df = pd.read_csv("formatted_output_Part_1.csv")

In [218]:
df.head()

No.             Time     Status  Duration (Seconds)  Access Code  \
0    1  10/3/2023 17:49  Completed               11646          NaN   
1    2  10/3/2023 17:49  Completed                 113          NaN   
2    3  10/3/2023 17:49  Completed                 572          NaN   
3    4  10/3/2023 17:50  Completed                 365          NaN   
4    5  10/3/2023 17:51  Completed                  92          NaN   

   Email Address      IP Address        Country    Region  \
0            NaN  100.36.167.226  United States  Virginia   
1            NaN   98.169.26.134  United States  Virginia   
2            NaN    98.169.52.23  United States  Virginia   
3            NaN  69.255.182.253  United States  Virginia   
4            NaN  138.88.108.230  United States       NaN   

  City / Town / District  ...  Q12.Some familiarity  Q12.Average user  \
0                Fairfax  ...                   1.0               NaN   
1                Fairfax  ...                   NaN               NaN   
2         Greenway Downs  ...                   NaN               1.0   
3            Gainesville  ...                   1.0               NaN   
4                    NaN  ...                   NaN               1.0   

   Q12.Frequent use for projects  Q12.Fluent/expert  Q13.Yes  Q13.No  Q14.Yes  \
0                            NaN                NaN      1.0     NaN      1.0   
1                            NaN                NaN      1.0     NaN      NaN   
2                            NaN                NaN      NaN     1.0      NaN   
3                            NaN                NaN      NaN     1.0      1.0   
4                            NaN                NaN      NaN     1.0      1.0   

   Q14.No Q14.NotSure                   Q15.GoalForLearningDataAnalytics  
0     NaN         NaN                                                NaN  
1     1.0         NaN                                                NaN  
2     NaN         1.0  As part of my work, I have access to a lot of ...  
3     NaN         NaN                                                NaN  
4     NaN         NaN                                              Learn  

[5 rows x 51 columns]

In [219]:
missing_values = df.isnull().sum()

In [220]:
missing_values

No.                                  0
Time                                 0
Status                               0
Duration (Seconds)                   0
Access Code                         36
Email Address                       36
IP Address                           0
Country                              0
Region                               4
City / Town / District               5
Latitude                             0
Longitude                            0
Q1.002                              16
Q1.004                              20
Q2.Male                             18
Q2.Female                           20
Q2.NonBinary                        35
Q2.PreferNotToAnswer                35
Q3.Age                               0
Q4.Height                            1
Q5.CountryOfCitizenship              2
Q6.UndergradDegree                   2
Q7.ExpectedGraduation                3
Q8.MicrosoftWindows                 12
Q8.AppleMacBook                     25
Q8.Other                 

In [221]:
columns_with_missing = df.columns[df.isnull().any()].tolist()

In [222]:
columns_with_missing

['Access Code',
 'Email Address',
 'Region',
 'City / Town / District',
 'Q1.002',
 'Q1.004',
 'Q2.Male',
 'Q2.Female',
 'Q2.NonBinary',
 'Q2.PreferNotToAnswer',
 'Q4.Height',
 'Q5.CountryOfCitizenship',
 'Q6.UndergradDegree',
 'Q7.ExpectedGraduation',
 'Q8.MicrosoftWindows',
 'Q8.AppleMacBook',
 'Q8.Other',
 'Unnamed: 26',
 'Q9.YesFullTime',
 'Q9.WorkingButNotFullTime',
 'Q9.NotWorking',
 'Q10.Little/None',
 'Q10.SomeFamiliarity',
 'Q10.AverageUser',
 'Q10.FrequentUseForproj',
 'Q10.FluentExpert',
 'Q11.Little/none',
 'Q11.Some familiarity',
 'Q11.Average user',
 'Q11.Frequent use for projects',
 'Q11.Fluent/expert',
 'Q12.Little/none',
 'Q12.Some familiarity',
 'Q12.Average user',
 'Q12.Frequent use for projects',
 'Q12.Fluent/expert',
 'Q13.Yes',
 'Q13.No',
 'Q14.Yes',
 'Q14.No',
 'Q14.NotSure',
 'Q15.GoalForLearningDataAnalytics']

In [223]:
new_df = df.copy()

In [224]:
def clean_data(new_df):
    new_df['Q1.002'] = new_df['Q1.002'].fillna(0)
    new_df['Q1.004'] = new_df['Q1.004'].fillna(0)
    
    new_df['Access Code'] = new_df['Access Code'].fillna('Unknown')
    new_df['Email Address'] = new_df['Email Address'].fillna('Unknown')
    
    new_df['Region'] = new_df['Region'].fillna('Unknown')
    new_df['City / Town / District'] = new_df['City / Town / District'].fillna('Unknown')
    
    gender_columns = ['Q2.Male', 'Q2.Female', 'Q2.NonBinary', 'Q2.PreferNotToAnswer']
    new_df[gender_columns] = new_df[gender_columns].fillna(0)

    new_df['Q3.Age'] = pd.to_numeric(new_df['Q3.Age'].str.replace(' ', ''), errors='coerce')

   

    new_df.loc[27, 'Q4.Height'] = 61
    new_df.loc[32, 'Q4.Height'] = 70

    new_df['Q5.CountryOfCitizenship'] = new_df['Q5.CountryOfCitizenship'].str.title()


    redundancy_map = {
    'Us': 'United States',
    'Usa': 'United States',
    'United States ': 'United States', 
    'United States ; Pakistan ': 'United States'  
    }

    new_df['Q5.CountryOfCitizenship'] = new_df['Q5.CountryOfCitizenship'].replace(redundancy_map)

    new_df['Q6.UndergradDegree'] = new_df['Q6.UndergradDegree'].str.title()


    personal_columns = ['Q4.Height', 'Q5.CountryOfCitizenship', 'Q6.UndergradDegree', 'Q7.ExpectedGraduation']
    new_df[personal_columns] = new_df[personal_columns].fillna('Not Provided')
    
    device_columns = ['Q8.MicrosoftWindows', 'Q8.AppleMacBook', 'Q8.Other']
    new_df[device_columns] = new_df[device_columns].fillna(0)

    
    
    if 'Unnamed: 26' in df.columns:
        new_df.drop(columns=['Unnamed: 26'], inplace=True)
    
    survey_columns = ['Q9.YesFullTime', 'Q9.WorkingButNotFullTime', 'Q9.NotWorking', 'Q10.Little/None', 
                      'Q10.SomeFamiliarity', 'Q10.AverageUser', 'Q10.FrequentUseForproj', 'Q10.FluentExpert', 
                      'Q11.Little/none', 'Q11.Some familiarity', 'Q11.Average user', 'Q11.Frequent use for projects', 
                      'Q11.Fluent/expert', 'Q12.Little/none', 'Q12.Some familiarity', 'Q12.Average user', 
                      'Q12.Frequent use for projects', 'Q12.Fluent/expert', 'Q13.Yes', 'Q13.No', 
                      'Q14.Yes', 'Q14.No', 'Q14.NotSure']
    new_df[survey_columns] = new_df[survey_columns].fillna(0)
    
    new_df['Q15.GoalForLearningDataAnalytics'] = new_df['Q15.GoalForLearningDataAnalytics'].fillna('Not Provided')
    
    return new_df

cleaned_df = clean_data(new_df)

In [225]:
cleaned_df

No.             Time      Status  Duration (Seconds) Access Code  \
0     1  10/3/2023 17:49   Completed               11646     Unknown   
1     2  10/3/2023 17:49   Completed                 113     Unknown   
2     3  10/3/2023 17:49   Completed                 572     Unknown   
3     4  10/3/2023 17:50   Completed                 365     Unknown   
4     5  10/3/2023 17:51   Completed                  92     Unknown   
5     6  10/3/2023 17:59  Incomplete                 221     Unknown   
6     7  10/3/2023 18:23   Completed              115983     Unknown   
7     8  10/3/2023 19:10  Incomplete                  40     Unknown   
8     9  10/3/2023 19:25   Completed                 137     Unknown   
9    10  10/3/2023 19:56  Incomplete                 179     Unknown   
10   11  10/3/2023 20:14   Completed                 127     Unknown   
11   12  10/3/2023 21:38   Completed                 269     Unknown   
12   13  10/3/2023 21:41   Completed                 117     Unknown   
13   14  10/3/2023 22:49   Completed                5966     Unknown   
14   15  10/3/2023 23:25   Completed                 254     Unknown   
15   16   10/4/2023 0:28   Completed                 526     Unknown   
16   17   10/4/2023 5:22   Completed                 398     Unknown   
17   18   10/4/2023 5:36   Completed              128253     Unknown   
18   19   10/4/2023 5:47   Completed                 239     Unknown   
19   20  10/4/2023 10:32   Completed                 255     Unknown   
20   21  10/4/2023 11:03   Completed                 256     Unknown   
21   22  10/4/2023 12:34   Completed                 674     Unknown   
22   23  10/4/2023 13:36   Completed                 130     Unknown   
23   24  10/4/2023 13:50   Completed                 275     Unknown   
24   25  10/4/2023 21:28   Completed                 325     Unknown   
25   26   10/5/2023 9:37   Completed                 257     Unknown   
26   27   10/5/2023 9:36   Completed                 657     Unknown   
27   28  10/5/2023 10:36   Completed                1750     Unknown   
28   29  10/5/2023 11:02   Completed                 887     Unknown   
29   30  10/5/2023 11:17   Completed                 180     Unknown   
30   31  10/5/2023 12:52   Completed                 392     Unknown   
31   32  10/5/2023 12:56  Incomplete                 192     Unknown   
32   33  10/5/2023 13:52   Completed                 277     Unknown   
33   34  10/5/2023 15:34   Completed                 119     Unknown   
34   35  10/5/2023 15:41   Completed                 149     Unknown   
35   36  10/5/2023 17:58   Completed                 285     Unknown   

   Email Address       IP Address        Country    Region  \
0        Unknown   100.36.167.226  United States  Virginia   
1        Unknown    98.169.26.134  United States  Virginia   
2        Unknown     98.169.52.23  United States  Virginia   
3        Unknown   69.255.182.253  United States  Virginia   
4        Unknown   138.88.108.230  United States   Unknown   
5        Unknown    108.56.134.80  United States  Virginia   
6        Unknown   108.48.175.220  United States  Virginia   
7        Unknown    173.67.199.85  United States  Virginia   
8        Unknown     74.96.114.12  United States  Virginia   
9        Unknown    107.77.241.32  United States     Texas   
10       Unknown   138.88.182.245  United States  Maryland   
11       Unknown     72.83.246.60  United States  Virginia   
12       Unknown    96.241.43.105  United States  Virginia   
13       Unknown    71.191.72.223  United States  Virginia   
14       Unknown    73.134.53.227  United States  Maryland   
15       Unknown     73.86.219.52  United States  Maryland   
16       Unknown   69.138.215.127  United States  Virginia   
17       Unknown    72.196.231.70  United States  Virginia   
18       Unknown     100.15.54.17  United States  Virginia   
19       Unknown   184.170.83.239  United States  Virginia   
20       Unknown   100.36.243.102  Un

### Discussion: 

#### To handle missing values in the dataset, each column was examined based on its content and purpose. For identifier and contact details such as 'Access Code' and 'Email Address', missing values were filled with 'Unknown' to avoid making incorrect assumptions. Geographic information like 'Region' and 'City / Town / District' was also filled with 'Unknown' due to the wide variability and lack of a logical mean or mode for imputation. Columns with categorical data, like 'Q5.CountryOfCitizenship', were filled with 'Not Provided' to maintain data integrity without introducing biases. In numeric survey response columns (e.g., Q8, Q9), missing values were replaced with 0, reflecting non-participation or zero count, which aligns with the survey's intent.

#### General data cleaning involved removing white spaces, standardizing casing, and consolidating redundant entries to ensure consistency and readability. This approach prioritizes transparency, avoiding assumptions that could skew the data. By using placeholders like 'Unknown' or 'Not Provided', the dataset maintains clarity about missing data while staying prepared for further analysis or modeling. Numeric fills such as 0 were used where appropriate, preserving the logical structure of the data without altering its distribution significantly.







In [227]:
# we can drop access code and email columns since they are entirely empty!

In [228]:
cleaned_df.drop(['Access Code','Email Address'], axis=1, inplace=True)

In [229]:
cleaned_df.head()

No.             Time     Status  Duration (Seconds)      IP Address  \
0    1  10/3/2023 17:49  Completed               11646  100.36.167.226   
1    2  10/3/2023 17:49  Completed                 113   98.169.26.134   
2    3  10/3/2023 17:49  Completed                 572    98.169.52.23   
3    4  10/3/2023 17:50  Completed                 365  69.255.182.253   
4    5  10/3/2023 17:51  Completed                  92  138.88.108.230   

         Country    Region City / Town / District  Latitude  Longitude  ...  \
0  United States  Virginia                Fairfax  -77.2891    38.8209  ...   
1  United States  Virginia                Fairfax  -77.2630    38.8605  ...   
2  United States  Virginia         Greenway Downs  -77.2331    38.9645  ...   
3  United States  Virginia            Gainesville  -77.6308    38.8175  ...   
4  United States   Unknown                Unknown  -97.8220    37.7510  ...   

   Q12.Some familiarity  Q12.Average user  Q12.Frequent use for projects  \
0                   1.0               0.0                            0.0   
1                   0.0               0.0                            0.0   
2                   0.0               1.0                            0.0   
3                   1.0               0.0                            0.0   
4                   0.0               1.0                            0.0   

   Q12.Fluent/expert  Q13.Yes  Q13.No  Q14.Yes Q14.No Q14.NotSure  \
0                0.0      1.0     0.0      1.0    0.0         0.0   
1                0.0      1.0     0.0      0.0    1.0         0.0   
2                0.0      0.0     1.0      0.0    0.0         1.0   
3                0.0      0.0     1.0      1.0    0.0         0.0   
4                0.0      0.0     1.0      1.0    0.0         0.0   

                    Q15.GoalForLearningDataAnalytics  
0                                       Not Provided  
1                                       Not Provided  
2  As part of my work, I have access to a lot of ...  
3                                       Not Provided  
4                                              Learn  

[5 rows x 48 columns]

In [230]:
output_filename='cleaned_output_Part_2.csv'
cleaned_df.to_csv(output_filename, index=False)

print(f"Cleaned data saved to {output_filename}")

Cleaned data saved to cleaned_output_Part_2.csv


## Part-3 (30 points)

#### Use the cleaned data from Part 2, prepare statistical summaries (counts, mean, unique etc.) for Questions Q1-Q14. You can just use describe() function in Pandas for this task. You can ignore Q15 as it has text. If you like to work on Q15, you are welcome to do so. Tell the useful story/insights for each question based on statistical summary from this survey data. Use the 580SurveyQuestion.pdf file to review the actual question of the survey and use it in your insights to make it more engaging.

In [233]:
df = pd.read_csv('cleaned_output_Part_2.csv')

In [234]:
columns = df.columns[10:47]

for col in columns:
    print(f"Statistical Summary for {col}:\n")
    print(df[col].describe(include='all'))
    print("\n" + "-" * 50 + "\n")

Statistical Summary for Q1.002:

count    36.000000
mean      0.555556
std       0.503953
min       0.000000
25%       0.000000
50%       1.000000
75%       1.000000
max       1.000000
Name: Q1.002, dtype: float64

--------------------------------------------------

Statistical Summary for Q1.004:

count    36.000000
mean      0.444444
std       0.503953
min       0.000000
25%       0.000000
50%       0.000000
75%       1.000000
max       1.000000
Name: Q1.004, dtype: float64

--------------------------------------------------

Statistical Summary for Q2.Male:

count    36.000000
mean      0.500000
std       0.507093
min       0.000000
25%       0.000000
50%       0.500000
75%       1.000000
max       1.000000
Name: Q2.Male, dtype: float64

--------------------------------------------------

Statistical Summary for Q2.Female:

count    36.000000
mean      0.444444
std       0.503953
min       0.000000
25%       0.000000
50%       0.000000
75%       1.000000
max       1.000000
Name: Q2.

## Results Interpretation:

### Q1: AIT-580 Section (Q1.002, Q1.004)

Survey Question: "AIT-580 Section? 002, 004"

Insight: A slight majority (55.6%) of the respondents are from Section 002, with the remaining 44.4% from Section 004. This nearly equal distribution suggests a balanced participation from both sections.

### Q2: Gender (Q2.Male, Q2.Female, Q2.NonBinary, Q2.PreferNotToAnswer)

Survey Question: "Gender? Male, Female, NonBinary, Prefer not to answer"

Insight: Half of the respondents identify as male (50%), followed by females (44.4%), with a very small percentage identifying as non-binary or preferring not to disclose. This indicates a predominantly binary gender representation.

### Q3: Age (Q3.Age)

Survey Question: "Age (years)"

Insight: The mean age of 81.67 and a maximum of 1993 suggest errors in the age data, possibly from incorrect data entry. Excluding outliers, ages cluster around the early to mid-20s, reflecting the typical age range of graduate students.

### Q4: Height (Q4.Height)

Survey Question: "Height (Inches)"

Insight: Heights vary significantly, with 70 inches being the most common. This data reflects a broad distribution of heights, possibly indicating a diverse group of participants.

### Q5: Country of Citizenship (Q5.CountryOfCitizenship)

Survey Question: "Country of Citizenship"

Insight: Most respondents are from the United States (28%), with others from 14 different countries, highlighting the international diversity of the cohort.

### Q6: Undergraduate Degree (Q6.UndergradDegree)

Survey Question: "Undergraduate Degree"

Insight: There is a wide variety of undergraduate degrees, with Economics being the most frequent (11%). This suggests that students from diverse academic backgrounds are pursuing data analytics.

### Q7: Expected Graduation Date (Q7.ExpectedGraduation)

Survey Question: "Expected Graduation date from Mason MS program?"

Insight: Graduation dates are varied, with "12/31/2021" being the most frequent response, reflecting the different timelines students are on to complete their studies.

### Q8: Type of Laptop Used (Q8.MicrosoftWindows, Q8.AppleMacBook, Q8.Other)

Survey Question: "Type of Laptop Used? Microsoft/Windows, Apple/MacBook, Other"

Insight: A majority use Microsoft Windows (64%), followed by Apple MacBooks (33%). This indicates a preference for mainstream operating systems among students, with a small fraction using alternatives.

### Q9: Employment Status (Q9.YesFullTime, Q9.WorkingButNotFullTime, Q9.NotWorking)

Survey Question: "Are you employed full-time while attending Mason? Yes, Full Time; Working, but not Full Time; Not Working"

Insight: About 31% of students work full-time, 14% work part-time, and nearly half (47%) are not working. This suggests a significant portion of students are fully dedicating their time to studies.

### Q10: Python Skill Level (Q10.Little/None to Q10.FluentExpert)

Survey Question: "What is your approximate current level of programming skill in Python?"

Insight: Most respondents are average users (42%), with few identifying as fluent experts. This suggests a moderate level of comfort with Python, but also room for skill improvement.

### Q11: R Skill Level (Q11.Little/none to Q11.Fluent/expert)

Survey Question: "What is your approximate current level of programming skill in R?"

Insight: Skills in R are mostly basic or intermediate, with only a few frequent users and very few experts. This shows a similar pattern of moderate familiarity as seen with Python.

### Q12: SQL Skill Level (Q12.Little/none to Q12.Fluent/expert)

Survey Question: "What is your approximate current level of programming skill in SQL?"

Insight: SQL skills also skew towards the lower end of proficiency, with no respondents claiming expert status. This highlights a potential area for skill enhancement.

### Q13: GitHub Use (Q13.Yes, Q13.No)

Survey Question: "Do you use GitHub for code repository and management?"

Insight: A third of respondents use GitHub, while the majority (67%) do not. This suggests that many students may not be leveraging GitHub for their coding projects or may lack familiarity with version control practices.

### Q14: Career in Data Science/Data Analytics (Q14.Yes, Q14.No, Q14.NotSure)

Survey Question: "Will you pursue your career in Data Science/Data Analytics field?"

Insight: Nearly half (47%) are inclined towards a career in data science, with a quarter undecided. This reflects strong interest, though some uncertainty remains among participants.
